template/src/components/Toast.tsx

```diff
@@ -1,202 +1,175 @@
-import React from 'react'
-import {
-  Animated,
-  StyleProp,
-  StyleSheet,
-  Text,
-  TextProps,
-  TouchableOpacity,
-  ViewProps,
-  ViewStyle,
-} from 'react-native'
-import {FontSizes, colors, metrics} from '../themes'
+import React, {useEffect, useRef, useState} from 'react'
+import {Animated, LayoutAnimation, StyleSheet, Text, TouchableOpacity, View} from 'react-native'
+import {FontSizes, colors, hitSlop, metrics} from '../themes'
 import Emitter from '../utilities/Emitter'
-import {getStatusBarHeight} from '../utilities/utils'
-
-const HEIGHT = getStatusBarHeight() + metrics.marginVertical
-
-type IToastProps = object
+import {useSafeAreaInsets} from 'react-native-safe-area-context'
+
+export const TOAST_TYPE = {
+  SUCCESS: 'SUCCESS',
+  ERROR: 'ERROR',
+  INFO: 'INFO',
+  WARNING: 'WARNING',
+} as const
+const TOAST_MESSAGE = 'SHOW_TOAST_MESSAGE'
```

Comment:
```suggestion
const TOAST_EVENT = {
showToastMessage: 'SHOW_TOAST_MESSAGE'
}
```

Reply:
I don't think we need this, currently we only have 1 purpose for this component => to show Toast
Maybe we will update this when we add more features for this Toast

Reply:
I suggest using TOAST_EVENTS because it makes things clearer than TOAST_MESSAGE. Even if we add a new event, we will know where to add it.

Reply:
How about I change it to
```suggestion
const SHOW_TOAST_MESSAGE = 'SHOW_TOAST_MESSAGE'
```
Or can you give me some examples for another `TOAST_EVENTS` (since I can only think of 1 purpose for this Component)

Reply:
You don't need to brainstorm other events. What I mean is we need a place to define the events.
You can change it to SHOW_TOAST_MESSAGE and put comment what it is.

Reply:
Okay, I will update this